cpp/src/cider/exec/nextgen/context/CodegenContext.cpp

```diff
@@ -291,5 +293,53 @@ void setArrowArrayLength(jitlib::JITValuePointer& arrow_array,
                                 .params_vector = {arrow_array.get(), len.get()}});
 }
 
+jitlib::JITValuePointer allocateArrowArrayBuffer(jitlib::JITValuePointer& arrow_array,
```

Comment:
Should we replace the `allocateArrowArrayBuffer(jitlib::JITValuePointer& arrow_array,
                                                 int64_t index,
                                                 jitlib::JITValuePointer& bytes)` in existing code?